java/ql/lib/semmle/code/java/security/TrustBoundaryViolationQuery.qll

```diff
@@ -0,0 +1,80 @@
+/** Provides classes and predicates to reason about trust boundary violations */
+
+import java
+private import semmle.code.java.dataflow.DataFlow
+private import semmle.code.java.controlflow.Guards
+private import semmle.code.java.dataflow.ExternalFlow
+private import semmle.code.java.dataflow.FlowSources
+private import semmle.code.java.frameworks.owasp.Esapi
+
+/**
+ * A source of data that crosses a trust boundary.
+ */
+abstract class TrustBoundaryViolationSource extends DataFlow::Node { }
+
+/**
+ * A node representing a servlet request.
+ */
+private class ServletRequestSource extends TrustBoundaryViolationSource {
+  ServletRequestSource() { this.asExpr().getType() instanceof HttpServletRequest }
```

Comment:
Shouldn't this just be a `RemoteFlowSource`? Any untrusted data can potentially be a problem in this case. Or is there something specific to a `HttpServletRequest` that you want to model here?

Reply:
After looking this over, the things I care about are better modeled using MaD in the `javax.servlet.http` models.